File: tests/r/test-shape.R
test_that("n_obs matches expected", {
  skip_if_no_golden(SCX_PATH)
  expect_equal(nrow(scx_ad()$obs), EXPECTED_N_OBS)
})

test_that("n_vars matches expected", {
  skip_if_no_golden(SCX_PATH)
  expect_equal(nrow(scx_ad()$var), EXPECTED_N_VARS)
})

test_that("shape matches reference", {
  skip_if_no_golden(SCX_PATH, REF_PATH)
  expect_equal(nrow(scx_ad()$obs), nrow(ref_ad()$obs))
  expect_equal(nrow(scx_ad()$var), nrow(ref_ad()$var))
})

test_that("obs names match reference", {
  skip_if_no_golden(SCX_PATH, REF_PATH)
  expect_identical(scx_ad()$obs_names, ref_ad()$obs_names)
})

test_that("var names match reference", {
  skip_if_no_golden(SCX_PATH, REF_PATH)
  expect_identical(scx_ad()$var_names, ref_ad()$var_names)
})

File: tests/python/test_shape.py
from conftest import EXPECTED_N_OBS, EXPECTED_N_VARS


def test_shape_matches_reference(scx_adata, ref_adata):
    assert scx_adata.shape == ref_adata.shape


def test_n_obs(scx_adata):
    assert scx_adata.n_obs == EXPECTED_N_OBS


def test_n_vars(scx_adata):
    assert scx_adata.n_vars == EXPECTED_N_VARS


def test_obs_names_match(scx_adata, ref_adata):
    assert list(scx_adata.obs_names) == list(ref_adata.obs_names)


def test_var_names_match(scx_adata, ref_adata):
    assert list(scx_adata.var_names) == list(ref_adata.var_names)
